plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -53,6 +56,71 @@ public static function load() {
 		do_action( 'woocommerce_helper_loaded' );
 	}
 
+	/**
+	 * Remove all notes signaling an error with the WCCOM API, when the request was successful.
+	 */
+	protected static function remove_api_error_notice() {
+		try {
+			$data_store = \WC_Data_Store::load( 'admin-note' );
+		} catch ( Exception $e ) {
+			return;
+		}
+
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+
+		if ( ! empty( $note_ids ) ) {
+			foreach ( $note_ids as $note_id ) {
+				$note = new Note( $note_id );
+				if ( $note->get_id() ) {
+					$data_store->delete( $note );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Adds at most one note signaling that there was an error with the WCCOM API.
+	 */
+	protected static function add_api_error_notice() {
+		try {
+			$data_store = \WC_Data_Store::load( 'admin-note' );
+		} catch ( Exception $e ) {
+			return;
+		}
+
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+
+		if ( ! empty( $note_ids ) ) {
+			$current_notice_id = array_shift( $note_ids );
+
+			foreach ( $note_ids as $note_id ) {
+				$note = new Note( $note_id );
+				if ( $note->get_id() ) {
+					$data_store->delete( $note );
+				}
+			}
+
+			$note = new Note( $current_notice_id );
+		} else {
+			$note = new Note();
+		}
+
+		$note->set_props(
+			array(
+				'title'        => __( 'WooCommerce.com API failure', 'woocommerce' ),
+				'content'      => __( 'A request to the WooCommerce.com API has failed - subscriptions may not be listed correctly. Please retry in a few minutes by refreshing this page.', 'woocommerce' ),
```

Comment:
Would be great to confirm the wording here

Reply:
Maybe this would be slightly less technical?

```suggestion
				'title'        => __( 'We’re having trouble connecting to WooCommerce.com', 'woocommerce' ),
				'content'      => __( 'Some subscription data may be temporarily unavailable. Please refresh the page in a few minutes to try again.', 'woocommerce' ),
```